app/controllers/stories_controller.rb

```diff
@@ -1,10 +1,7 @@
 class StoriesController < ApplicationController
-  def index
-    @stories = Story.all
-  end
-
+  before_action :authenticate_user!
```

Comment:
Here's what I would suggest. Set the `root_path` to `controller: :stories, action: :index` (The same action you removed above. If you examine the `index`action on the `welcome_controller`, youäll see that it is well suited to be part of the `stories_controller`. (If you move that change, you can safely remove the `welcome_controller` 

In order to only 'lock' the `show` action to logged in users use `before_action :authenticate_user!, only: [:show]`